File: domain/src/main/java/com/arturogutierrez/openticator/domain/category/CategoryFactory.java
package com.arturogutierrez.openticator.domain.category;

import com.arturogutierrez.openticator.domain.category.model.Category;
import com.arturogutierrez.openticator.domain.category.model.EmptyCategory;
import java.util.UUID;
import javax.inject.Inject;

public class CategoryFactory {

  @Inject
  public CategoryFactory() {

  }

  public Category createCategory(String categoryName) {
    String categoryId = UUID.randomUUID().toString();
    return new Category(categoryId, categoryName);
  }

  public Category createEmptyCategory() {
    return new EmptyCategory();
  }
}

File: domain/src/main/kotlin/com/arturogutierrez/openticator/domain/category/CategoryFactory.kt
package com.arturogutierrez.openticator.domain.category

import com.arturogutierrez.openticator.domain.category.model.Category
import java.util.*
import javax.inject.Inject

class CategoryFactory @Inject constructor() {

  fun createCategory(categoryName: String): Category {
    val categoryId = UUID.randomUUID().toString()
    return Category(categoryId, categoryName)
  }

  fun createEmptyCategory(): Category {
    return Category.empty
  }
}
